### DISCUSSION TITLE
add topic to find pulsar packages on github

### DISCUSSION THREAD

**Discussion by sertonix** [pulsar-edit/.github#69] [Pulsar Features]
I think it would be nice to better allow packages to be installed from various sources.
One good way I think would be install directly from github. The problem is that currently it is hard to find packages on github. Therefor I think it would make sense to add a topic to all repositories.
My suggestion would be [`pulsar-edit-package`](https://github.com/topics/pulsar-edit-package).
This would also allow to distinguish the pulsar packages from pulsar related repos in the `pulsar-edit` org.

Edit: As suggested by @confused-Techie `pulsar-edit` and `pulsar-package` would be better as topic.

**Comment by Daeraxa** [comment]
Instead of a topic ~~why don't we just have a "Packages" category or something~~ we have a "packages" category in our discussions that anyone who wants to promote their packages, can, regardless of whether it is on the package repo or not which can be useful if people want to discuss "in dev" packages they don't want to release yet. Labels could be used to add additional info like "beta" or "release".

**Reply by sertonix** [reply]
Maybe the word "topic" is a bit misleading (I think it is). I mean the tags (github calls them topics) you can add to repositories like [`javascript`](https://github.com/topics/javascript).
The idea is a place to search for packages not promote them.

**Reply by Daeraxa** [reply]
oh I see, I've not even looked at that before. Are we talking about internal or community packages?

**Reply by sertonix** [reply]
Both. But the internal would most likely be the first

**Reply by Daeraxa** [reply]
The vast majority of our repos are packages with just a few meta repos, could we not just add "pulsar" and "pulsar-edit"? Its a little more obvious to community packages then as they are probably quite likely to add those.

**Reply by sertonix** [reply]
Yes that is true. `pulsar-edit` is better since `pulsar` is already used.
Maybe a member of pulsar could ["create this topic"](https://github.com/topics/pulsar-edit). It would allow us to add a description and images

**Reply by sertonix** [reply]
While searching through the repos I remembered this idea again.
I think there should be a topic specifically for packages since then you could use [this](https://github.com/search?q=org%3Apulsar-edit+archived%3Afalse+fork%3Atrue+topic%3Apulsar-edit-package&type=repositories) to find all the packages that pulsar-edit maintains. Not all repos that are related to pulsar

**Comment by confused-Techie** [comment]
I actually totally love this idea, and do think we should start using a pulsar-edit topic.

What if we do this?

`pulsar-edit` is the Keyword, topic (whatever) for anything related to pulsar, ie all of our repos 

Then `pulsar-package` is another keyword that can be added for any packages. Third party or first party?

**Reply by sertonix** [reply]
I would say first and third party packages. So it may be possible to search for pulsar packages on github in pulsar.

**Reply by confused-Techie** [reply]
Also slightly saving this for myself later. Even if we don't do this going forward we should probably take some ownership of `pulsar-edit` as a topic. Looks like this can be done with a PR [here](https://github.com/github/explore/blob/main/CONTRIBUTING.md).

But will try and take a look with any free time

**Reply by confused-Techie** [reply]
> I would say first and third party packages. So it may be possible to search for pulsar packages on github in pulsar.

Yeah exactly, no need to separate them even more

**Reply by sertonix** [reply]
Since pulsar is not a unique name I think it would make sense that packages should have both `pulsar-edit` and `pulsar-package`.

**Reply by confused-Techie** [reply]
Also it actually looks like no one on the team can add the topic themselves. The guidelines for creating a topic say "Avoid conflicts of interest. Maintainers of a project cannot add a topic or collection for their own project. If a topic is popular enough to warrant inclusion, someone else will add or improve it.".

Meaning we can't curate the topic or add things like a description or title. Someone else from the community will have to. But doesn't stop us from starting to use the topics across our repos

**Reply by sertonix** [reply]
Should I add issues to add the topics like [this](https://github.com/pulsar-edit/github/issues/6)?
Or should I just prepair a list?

**Reply by confused-Techie** [reply]
> Should I add issues to add the topics like [this](https://github.com/pulsar-edit/github/issues/6)? Or should I just prepair a list?

You can do what you're most comfortable with. If you'd rather just make a list that's fine by me, and I'd be happy to try and check in on everything, although am still behind on @Sertonix list

**Reply by sertonix** [reply]
| repo | `pulsar-edit` | `pulsar-package`
|---|---|---
| [`.github`](https://github.com/pulsar-edit/.github) | <li>[X] </li> | 
| [`action-pulsar-dependency`](https://github.com/pulsar-edit/action-pulsar-dependency) | <li>[X] </li> |
| [`active-editor-info`](https://github.com/pulsar-edit/active-editor-info) | <li>[X] </li> | <li>[X] </li>
| [`api`](https://github.com/pulsar-edit/api) | <li>[X] </li> |
| [`archive-view`](https://github.com/pulsar-edit/archive-view) | <li>[X] </li> | <li>[X] </li>
| [`archived-alternative-method`](https://github.com/pulsar-edit/archived-alternative-method) | <li>[X] </li> |
| [`ascii-art`](https://github.com/pulsar-edit/ascii-art) | <li>[ ] </li> | <li>[ ] </li>
| [`atom-babel6-transpiler`](https://github.com/pulsar-edit/atom-babel6-transpiler) | <li>[ ] </li> |
| [`atom-babel7-transpiler`](https://github.com/pulsar-edit/atom-babel7-transpiler) | <li>[ ] </li> |
| [`atom-bugs`](https://github.com/pulsar-edit/atom-bugs) | <li>[ ] </li> |
| [`atom-ide-base`](https://github.com/pulsar-edit/atom-ide-base) | <li>[ ] </li> | <li>[ ] </li>
| [`atom-ide-code-format-1`](https://github.com/pulsar-edit/atom-ide-code-format-1) | <li>[ ] </li> | <li>[ ] </li>
| [`atom-ide-console`](https://github.com/pulsar-edit/atom-ide-console) | <li>[ ] </li> | <li>[ ] </li>
| [`atom-ide-datatip`](https://github.com/pulsar-edit/atom-ide-datatip) | <li>[ ] </li> | <li>[ ] </li>
| [`atom-ide-debugger`](https://github.com/pulsar-edit/atom-ide-debugger) | <li>[ ] </li> | <li>[ ] </li>
| [`atom-ide-definitions`](https://github.com/pulsar-edit/atom-ide-definitions) | <li>[ ] </li> | <li>[ ] </li>
| [`atom-ide-diagnostics`](https://github.com/pulsar-edit/atom-ide-diagnostics) | <li>[ ] </li> | <li>[ ] </li>
| [`atom-ide-hyperclick-1`](https://github.com/pulsar-edit/atom-ide-hyperclick-1) | <li>[ ] </li> | <li>[ ] </li>
| [`atom-ide-javascript`](https://github.com/pulsar-edit/atom-ide-javascript) | <li>[ ] </li> | <li>[ ] </li>
| [`atom-ide-markdown-service`](https://github.com/pulsar-edit/atom-ide-markdown-service) | <li>[ ] </li> | <li>[ ] </li>
| [`atom-ide-signature-help`](https://github.com/pulsar-edit/atom-ide-signature-help) | <li>[ ] </li> | <li>[ ] </li>
| [`atom-ide-template`](https://github.com/pulsar-edit/atom-ide-template) | <li>[ ] </li> |
| [`atom-ide-template-js`](https://github.com/pulsar-edit/atom-ide-template-js) | <li>[ ] </li> |
| [`atom-ide-vue`](https://github.com/pulsar-edit/atom-ide-vue) | <li>[ ] </li> | <li>[ ] </li>
| [`atom-keymap`](https://github.com/pulsar-edit/atom-keymap) | <li>[ ] </li> |
| [`atom-languageclient`](https://github.com/pulsar-edit/atom-languageclient) | <li>[ ] </li> | <li>[ ] </li>
| [`atom-mocha-test-runner`](https://github.com/pulsar-edit/atom-mocha-test-runner) | <li>[ ] </li> |
| [`atom-package-downloader`](https://github.com/pulsar-edit/atom-package-downloader) | <li>[ ] </li> |
| [`atom-select-list`](https://github.com/pulsar-edit/atom-select-list) | <li>[ ] </li> |
| [`atom-selector-linter`](https://github.com/pulsar-edit/atom-selector-linter) | <li>[ ] </li> |
| [`atom-ternjs`](https://github.com/pulsar-edit/atom-ternjs) | <li>[ ] </li> |
| [`atom-trello`](https://github.com/pulsar-edit/atom-trello) | <li>[ ] </li> |
| [`autocomplete-atom-api`](https://github.com/pulsar-edit/autocomplete-atom-api) | <li>[ ] </li> | <li>[ ] </li>
| [`autocomplete-emojis`](https://github.com/pulsar-edit/autocomplete-emojis) | <li>[ ] </li> | <li>[ ] </li>
| [`autosave`](https://github.com/pulsar-edit/autosave) | <li>[ ] </li> | <li>[ ] </li>
| [`avm`](https://github.com/pulsar-edit/avm) | <li>[ ] </li> |
| [`babel-plugin-chai-assert-async`](https://github.com/pulsar-edit/babel-plugin-chai-assert-async) | <li>[ ] </li> |
| [`biohazard-alert`](https://github.com/pulsar-edit/biohazard-alert) | <li>[ ] </li> |
| [`bracket-matcher`](https://github.com/pulsar-edit/bracket-matcher) | <li>[ ] </li> | <li>[ ] </li>
| [`ci`](https://github.com/pulsar-edit/ci) | <li>[ ] </li> |
| [`clear-cut`](https://github.com/pulsar-edit/clear-cut) | <li>[ ] </li> |
| [`core-languages`](https://github.com/pulsar-edit/core-languages) | <li>[ ] </li> |
| [`delegato`](https://github.com/pulsar-edit/delegato) | <li>[ ] </li> |
| [`development-images`](https://github.com/pulsar-edit/development-images) | <li>[ ] </li> |
| [`donna`](https://github.com/pulsar-edit/donna) | <li>[ ] </li> |
| [`editor-background`](https://github.com/pulsar-edit/editor-background) | <li>[ ] </li> | <li>[ ] </li>
| [`editor-stats`](https://github.com/pulsar-edit/editor-stats) | <li>[ ] </li> | <li>[ ] </li>
| [`electron-link`](https://github.com/pulsar-edit/electron-link) | <li>[ ] </li> |
| [`essential-packages`](https://github.com/pulsar-edit/essential-packages) | <li>[ ] </li> |
| [`etch`](https://github.com/pulsar-edit/etch) | <li>[ ] </li> |
| [`event-kit`](https://github.com/pulsar-edit/event-kit) | <li>[ ] </li> |
| [`find-and-replace`](https://github.com/pulsar-edit/find-and-replace) | <li>[ ] </li> | <li>[ ] </li>
| [`first-mate`](https://github.com/pulsar-edit/first-mate) | <li>[ ] </li> |
| [`fs-admin`](https://github.com/pulsar-edit/fs-admin) | <li>[ ] </li> |
| [`fs-plus`](https://github.com/pulsar-edit/fs-plus) | <li>[ ] </li> |
| [`fuzzaldrin`](https://github.com/pulsar-edit/fuzzaldrin) | <li>[ ] </li> |
| [`fuzzy-finder`](https://github.com/pulsar-edit/fuzzy-finder) | <li>[ ] </li> | <li>[ ] </li>
| [`fuzzy-native`](https://github.com/pulsar-edit/fuzzy-native) | <li>[ ] </li> |
| [`git-utils`](https://github.com/pulsar-edit/git-utils) | <li>[ ] </li> |
| [`github`](https://github.com/pulsar-edit/github) | <li>[x] </li> | <li>[x] </li>
| [`grim`](https://github.com/pulsar-edit/grim) | <li>[ ] </li> |
| [`grunt-atomdoc`](https://github.com/pulsar-edit/grunt-atomdoc) | <li>[ ] </li> |
| [`ide-csharp`](https://github.com/pulsar-edit/ide-csharp) | <li>[ ] </li> | <li>[ ] </li>
| [`ide-go`](https://github.com/pulsar-edit/ide-go) | <li>[ ] </li> | <li>[ ] </li>
| [`ide-java`](https://github.com/pulsar-edit/ide-java) | <li>[ ] </li> | <li>[ ] </li>
| [`ide-json`](https://github.com/pulsar-edit/ide-json) | <li>[ ] </li> | <li>[ ] </li>
| [`ide-php`](https://github.com/pulsar-edit/ide-php) | <li>[ ] </li> | <li>[ ] </li>
| [`ide-python`](https://github.com/pulsar-edit/ide-python) | <li>[ ] </li> | <li>[ ] </li>
| [`ide-sourcekit`](https://github.com/pulsar-edit/ide-sourcekit) | <li>[ ] </li> | <li>[ ] </li>
| [`ide-typescript`](https://github.com/pulsar-edit/ide-typescript) | <li>[ ] </li> | <li>[ ] </li>
| [`ide-yaml`](https://github.com/pulsar-edit/ide-yaml) | <li>[ ] </li> | <li>[ ] </li>
| [`jasmine-focused`](https://github.com/pulsar-edit/jasmine-focused) | <li>[ ] </li> |
| [`jasmine-json`](https://github.com/pulsar-edit/jasmine-json) | <li>[ ] </li> |
| [`jasmine-tagged`](https://github.com/pulsar-edit/jasmine-tagged) | <li>[ ] </li> |
| [`joanna`](https://github.com/pulsar-edit/joanna) | <li>[ ] </li> |
| [`js-git-utils`](https://github.com/pulsar-edit/js-git-utils) | <li>[ ] </li> |
| [`key-path-helpers`](https://github.com/pulsar-edit/key-path-helpers) | <li>[ ] </li> |
| [`keybinding-resolver`](https://github.com/pulsar-edit/keybinding-resolver) | <li>[ ] </li> | <li>[ ] </li>
| [`keyboard-layout`](https://github.com/pulsar-edit/keyboard-layout) | <li>[ ] </li> |
| [`language-examples`](https://github.com/pulsar-edit/language-examples) | <li>[ ] </li> | <li>[ ] </li>
| [`language-pegjs`](https://github.com/pulsar-edit/language-pegjs) | <li>[ ] </li> | <li>[ ] </li>
| [`language-puppet`](https://github.com/pulsar-edit/language-puppet) | <li>[ ] </li> | <li>[ ] </li>
| [`less-cache`](https://github.com/pulsar-edit/less-cache) | <li>[ ] </li> |
| [`line-top-index`](https://github.com/pulsar-edit/line-top-index) | <li>[ ] </li> |
| [`linter-languagetool`](https://github.com/pulsar-edit/linter-languagetool) | <li>[ ] </li> | <li>[ ] </li>
| [`loophole`](https://github.com/pulsar-edit/loophole) | <li>[ ] </li> |
| [`mistaken-pull-closer`](https://github.com/pulsar-edit/mistaken-pull-closer) | <li>[ ] </li> |
| [`mixto`](https://github.com/pulsar-edit/mixto) | <li>[ ] </li> |
| [`neon`](https://github.com/pulsar-edit/neon) | <li>[ ] </li> |
| [`neon-cli`](https://github.com/pulsar-edit/neon-cli) | <li>[ ] </li> |
| [`node`](https://github.com/pulsar-edit/node) | <li>[ ] </li> |
| [`node-ctags`](https://github.com/pulsar-edit/node-ctags) | <li>[ ] </li> |
| [`node-keytar`](https://github.com/pulsar-edit/node-keytar) | <li>[ ] </li> |
| [`node-ls-archive`](https://github.com/pulsar-edit/node-ls-archive) | <li>[ ] </li> |
| [`node-nslog`](https://github.com/pulsar-edit/node-nslog) | <li>[ ] </li> |
| [`node-oniguruma`](https://github.com/pulsar-edit/node-oniguruma) | <li>[ ] </li> |
| [`node-pathwatcher`](https://github.com/pulsar-edit/node-pathwatcher) | <li>[ ] </li> |
| [`node-spellchecker`](https://github.com/pulsar-edit/node-spellchecker) | <li>[ ] </li> |
| [`notebook`](https://github.com/pulsar-edit/notebook) | <li>[ ] </li> | <li>[ ] </li>
| [`notifications`](https://github.com/pulsar-edit/notifications) | <li>[ ] </li> | <li>[ ] </li>
| [`nsfw`](https://github.com/pulsar-edit/nsfw) | <li>[ ] </li> |
| [`package-backend`](https://github.com/pulsar-edit/package-backend) | <li>[x] </li> |
| [`package-frontend`](https://github.com/pulsar-edit/package-frontend) | <li>[ ] </li> |
| [`ppm`](https://github.com/pulsar-edit/ppm) | <li>[ ] </li> |
| [`pr-changelog`](https://github.com/pulsar-edit/pr-changelog) | <li>[ ] </li> |
| [`prettier-config-atomic`](https://github.com/pulsar-edit/prettier-config-atomic) | <li>[ ] </li> |
| [`project-ring`](https://github.com/pulsar-edit/project-ring) | <li>[ ] </li> | <li>[ ] </li>
| [`property-accessors`](https://github.com/pulsar-edit/property-accessors) | <li>[ ] </li> |
| [`pulsar`](https://github.com/pulsar-edit/pulsar) | <li>[x] </li> |
| [`pulsar-assets`](https://github.com/pulsar-edit/pulsar-assets) | <li>[ ] </li> |
| [`pulsar-edit.github.io`](https://github.com/pulsar-edit/pulsar-edit.github.io) | <li>[ ] </li> |
| [`pulsar-pathwatcher`](https://github.com/pulsar-edit/pulsar-pathwatcher) | <li>[ ] </li> |
| [`react`](https://github.com/pulsar-edit/react) | <li>[ ] </li> |
| [`repo-doc-gen`](https://github.com/pulsar-edit/repo-doc-gen) | <li>[ ] </li> |
| [`revert-buffer`](https://github.com/pulsar-edit/revert-buffer) | <li>[ ] </li> |
| [`rollup-plugin-atomic`](https://github.com/pulsar-edit/rollup-plugin-atomic) | <li>[ ] </li> |
| [`scandal`](https://github.com/pulsar-edit/scandal) | <li>[ ] </li> |
| [`scoped-property-store`](https://github.com/pulsar-edit/scoped-property-store) | <li>[ ] </li> |
| [`scroll-searcher`](https://github.com/pulsar-edit/scroll-searcher) | <li>[ ] </li> | <li>[ ] </li>
| [`scrollbar-style`](https://github.com/pulsar-edit/scrollbar-style) | <li>[ ] </li> |
| [`season`](https://github.com/pulsar-edit/season) | <li>[ ] </li> |
| [`selector-kit`](https://github.com/pulsar-edit/selector-kit) | <li>[ ] </li> |
| [`serializable`](https://github.com/pulsar-edit/serializable) | <li>[ ] </li> |
| [`service-hub`](https://github.com/pulsar-edit/service-hub) | <li>[ ] </li> |
| [`slick`](https://github.com/pulsar-edit/slick) | <li>[ ] </li> |
| [`snippets`](https://github.com/pulsar-edit/snippets) | <li>[ ] </li> | <li>[ ] </li>
| [`sort-lines`](https://github.com/pulsar-edit/sort-lines) | <li>[ ] </li> | <li>[ ] </li>
| [`span-skip-list`](https://github.com/pulsar-edit/span-skip-list) | <li>[ ] </li> |
| [`spell-check`](https://github.com/pulsar-edit/spell-check) | <li>[ ] </li> | <li>[ ] </li>
| [`superstring`](https://github.com/pulsar-edit/superstring) | <li>[ ] </li> |
| [`superstring-wasm`](https://github.com/pulsar-edit/superstring-wasm) | <li>[ ] </li> |
| [`symbols-view`](https://github.com/pulsar-edit/symbols-view) | <li>[ ] </li> | <li>[ ] </li>
| [`teletype`](https://github.com/pulsar-edit/teletype) | <li>[ ] </li> | <li>[ ] </li>
| [`teletype-client`](https://github.com/pulsar-edit/teletype-client) | <li>[ ] </li> |
| [`teletype-crdt`](https://github.com/pulsar-edit/teletype-crdt) | <li>[ ] </li> |
| [`teletype-server`](https://github.com/pulsar-edit/teletype-server) | <li>[ ] </li> |
| [`tello`](https://github.com/pulsar-edit/tello) | <li>[ ] </li> |
| [`template-syntax`](https://github.com/pulsar-edit/template-syntax) | <li>[ ] </li> |
| [`text-buffer`](https://github.com/pulsar-edit/text-buffer) | <li>[ ] </li> |
| [`timecop`](https://github.com/pulsar-edit/timecop) | <li>[ ] </li> | <li>[ ] </li>
| [`toggle-quotes`](https://github.com/pulsar-edit/toggle-quotes) | <li>[ ] </li> | <li>[ ] </li>
| [`town-crier`](https://github.com/pulsar-edit/town-crier) | <li>[ ] </li> | <li>[ ] </li>
| [`tree-sitter`](https://github.com/pulsar-edit/tree-sitter) | <li>[ ] </li> |
| [`tree-view`](https://github.com/pulsar-edit/tree-view) | <li>[ ] </li> | <li>[ ] </li>
| [`tree-view-finder`](https://github.com/pulsar-edit/tree-view-finder) | <li>[ ] </li> | <li>[ ] </li>
| [`tree-view-search-bar`](https://github.com/pulsar-edit/tree-view-search-bar) | <li>[ ] </li> | <li>[ ] </li>
| [`ui-theme-template`](https://github.com/pulsar-edit/ui-theme-template) | <li>[ ] </li> |
| [`underscore-plus`](https://github.com/pulsar-edit/underscore-plus) | <li>[ ] </li> |
| [`what-the-status`](https://github.com/pulsar-edit/what-the-status) | <li>[ ] </li> |
| [`whats-my-line`](https://github.com/pulsar-edit/whats-my-line) | <li>[ ] </li> |

**Reply by sertonix** [reply]
@confused-Techie is that list ok like that?
Would be interisting if you can also check them.
And a question would be if we really want to tag forks without updates with `pulsar-edit` like for example [project-ring](https://github.com/atom/project-ring)

**Reply by confused-Techie** [reply]
Yeah I tried to edit this as well, just so I could properly check them off as I go through them, but no dice. Might be worth it to make this an issue on `.github` repo, otherwise I'll work on this when I can. Thanks for compiling the list together.

As for tagging forks I think that depends. Are they published to the package registry as our own? If not then no need, but if they are then we should

**Reply by confused-Techie** [reply]
Also @Sertonix for one of these I made an archival notice as it's no longer supported [here](https://github.com/pulsar-edit/action-setup-pulsar/pull/1)

**Comment by SilentGlasses** [comment]
I am a bit confused about what this is doing other than listing things...     Are we just taking inventory/cataloging the available packages?

If this is just a topic that things can get tagged with, why not just Pulsar, like they did for [Atom](https://github.com/topics/atom)


If it is cataloging, a better way to do this would be to work with the creators for community packages to migrate their packages to work with Pulsar as well as make the needed naming changes... This way anyone can search for pulsar plugins/packages and packages get updated to work with the vision and goals of the project.

So instead of `atom-bugs` it should be `pulsar-bugs` and make the code changes as well.
